tests/bitarithm_timings/main.c

```diff
@@ -38,26 +40,29 @@
 #define TIMEOUT (TIMEOUT_S * US_PER_SEC)
 #define PER_ITERATION (4)
 
-static void callback(void *done_)
+static atomic_bool done;
+
+static void callback(void *unused)
 {
-    volatile int *done = done_;
-    *done = 1;
+    (void)unused;
+    atomic_store(&done, true);
 }
 
 static void run_test(const char *name, unsigned (*test)(unsigned))
 {
-    volatile int done = 0;
     unsigned i = 0;
     unsigned long count = 0;
+    atomic_store(&done, false);
 
-    xtimer_t xtimer;
-    xtimer.callback = callback;
-    xtimer.arg = (void *) &done;
-
+    xtimer_t xtimer = { .callback = callback };
     xtimer_set(&xtimer, TIMEOUT);
 
     do {
-        if (i++ == -1u) {
+        if (++i == UINT_MAX) {
```

Comment:
I just checked and only this change is necessary. The problem is that the change is provided in the same commit as the use C11 atomic. Just moving the latter in a separate commit would be better for me.

Reply:
Done